plugins/hydra_orion_sweeper/hydra_plugins/hydra_orion_sweeper/_impl.py

```diff
@@ -0,0 +1,335 @@
+                pass
+
+
+@contextmanager
+def clientctx(client):
+    try:
+        yield client
+    finally:
+        client.close()
+
+
+class OrionSweeperImpl(Sweeper):
+    def __init__(
+        self,
+        orion: OrionClientConf,
+        worker: WorkerConf,
+        algorithm: AlgorithmConf,
+        storage: StorageConf,
+        parametrization: Optional[DictConfig],
+    ):
+        self.orion_config = orion
+        self.worker_config = worker
+        self.algo_config = algorithm
+        self.storage_config = storage
+
+        self.launcher: Optional[Launcher] = None
+        self.hydra_context: Optional[HydraContext] = None
+        self.job_results = None
+        self.job_idx: Optional[int] = None
+
+        self.space_parser = SpaceParser()
+        self.space_parser.add_from_parametrization(parametrization)
+
+    def setup(
+        self,
+        *,
+        hydra_context: HydraContext,
+        task_function: TaskFunction,
+        config: DictConfig,
+    ) -> None:
+        self.job_idx = 0
+        self.config = config
+        self.hydra_context = hydra_context
+        self.pending_trials = set()
+
+        self.space = None
+        self.arguments = dict()
+
+        log.info("Starting launcher")
+
+        self.launcher = Plugins.instance().instantiate_launcher(
+            hydra_context=hydra_context, task_function=task_function, config=config
+        )
+
+    def suggest_trials(self, count) -> List[Trial]:
+        """Suggest a bunch of trials to be dispatched to the workers"""
+        trials = []
+
+        for _ in range(count):
+            try:
+                trial = self.client.suggest(pool_size=count)
+                trials.append(trial)
+
+            # non critical errors
+            except WaitingForTrials:
+                break
+
+            except ReservationRaceCondition:
+                break
+
+            except CompletedExperiment:
+                break
+
+        return trials
+
+    def new_experiment(self, arguments) -> ExperimentClient:
+        """Initialize orion client from the config and the arguments"""
+
+        self.space_parser.add_from_overrides(arguments)
+        self.space, self.arguments = self.space_parser.space()
+
+        dict_config = OmegaConf.to_container(self.algo_config)
+        algo_type = dict_config.pop("type", "random")
+        algo_config = dict_config.pop("config", dict())
+
+        return create_experiment(
+            name=self.orion_config.name,
+            version=self.orion_config.version,
+            space=self.space,
+            algorithms={algo_type: algo_config},
+            strategy=None,
+            max_trials=self.worker_config.max_trials,
+            max_broken=self.worker_config.max_broken,
+            storage=self.storage_config,
+            branching=self.orion_config.branching,
+            max_idle_time=None,
+            heartbeat=None,
+            working_dir=None,
```

Comment:
Supporting working_dir would be important. Algorithms like PBT rely on it. Does it cause any problem with Hydra?

Reply:
Hydra has its own working dir configuration, I haven't looked at it yet

Reply:
It would be good to link Oríon's working dir to it then.

Reply:
Forget it, it is not possible since trials may be resumed leading to different jobs in Hydra, and thus different job id and different working dir. We'll need to support working_dir in Oríon in parallel to working_dir in Hydra.